aws-iot-dimension/src/test/java/com/amazonaws/iot/dimension/CreateHandlerTest.java

```diff
@@ -0,0 +1,168 @@
+package com.amazonaws.iot.dimension;
+
+import static com.amazonaws.iot.dimension.TestConstants.CLIENT_REQUEST_TOKEN;
+import static com.amazonaws.iot.dimension.TestConstants.DESIRED_TAGS;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_ARN;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_NAME;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_TYPE;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_VALUE;
+import static com.amazonaws.iot.dimension.TestConstants.MODEL_TAGS;
+import static com.amazonaws.iot.dimension.TestConstants.SDK_MODEL_TAG;
+import static com.amazonaws.iot.dimension.TestConstants.SDK_SYSTEM_TAG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.services.iot.model.CreateDimensionRequest;
+import software.amazon.awssdk.services.iot.model.CreateDimensionResponse;
+import software.amazon.awssdk.services.iot.model.DimensionType;
+import software.amazon.awssdk.services.iot.model.ResourceAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+@ExtendWith(MockitoExtension.class)
+public class CreateHandlerTest {
```

Comment:
Maybe also add a test for create dimension throwing invalid request exception?

Reply:
This will be covered as part of the integration tests so looks good to have one exception handling test as part of unit test.